src/js/components/Marathon.jsx

```diff
@@ -44,21 +46,26 @@ var Marathon = React.createClass({
   },
 
   componentDidMount: function () {
-    this.onRouteChange();
+    PluginActions.requestPlugins();
 
+    this.onRouteChange();
     this.bindKeyboardShortcuts();
 
-    PluginStore.once(PluginEvents.BOOTSTRAP_COMPLETE, () => {
-      this.startPolling();
-    });
-
-    PluginStore.bootstrap();
+    PluginStore.on(PluginEvents.CHANGE, this.onPluginStoreChange);
   },
 
   componentWillReceiveProps: function () {
     this.onRouteChange();
   },
 
+  onPluginStoreChange: function () {
+    var pluginLoadingState = PluginStore.getPluginLoadingState();
```

Comment:
I'll add a helper function to determine the plugin loading state and refactor the respective conditions. 
Adding an `isComplete` getter to the PluginStore is problematic as complete in this case means every state but initial and loading is complete and in other cases could mean something different e.g. only if every plugin loads successfully the whole process is complete.